### PR TITLE
DOC: add instruction how to support new platform version

### PR DESCRIPTION
**Pull request by Undin**

### CODE REVIEW

File: MAINTAINING.md [hunk 1]
```diff
@@ -49,6 +49,68 @@ Note `--distribution-type all`.
...
+ 
+Step to step instruction how to support new platform version:
+* Drop all code related to the `old` platform version, i.e. drop `gradle-%old%.properties` and
+all `%module_name%/src/%old%` directories in each module
+* Move code from `%module_name%/src/%current%` directories into common source set,
+i.e. `%module_name%/src/%current%/main` into `%module_name%/src/main` and `%module_name%/src/%current%/test` into `%module_name%/src/test`.
+Also, simplify the moved code if possible.
+* Add support of `new` platform, i.e. add `gradle-%new%.properties` with all necessary properties and make it compile.
+It may be required to extract some code into platform-specific source sets to make plugin compile with each supported platform.
+See [Tips and tricks](#tips_and_tricks) section for the most common examples of how to do it
```
Comment (avrong) [inline MAINTAINING.md:77]:
```suggestion
See [Tips and tricks](#tips-and-tricks) section for the most common examples of how to do it
```

File: MAINTAINING.md [hunk 2]
```diff
@@ -49,6 +49,68 @@ Note `--distribution-type all`.
...
+* Fix tests
+* Update log path in run configurations like `runIDEA` and `runCLion`
+* Update CI workflows to use new platform version instead of old one
```
Comment (avrong) [inline MAINTAINING.md:80]: Are there any special CI-specific points to think about when upgrading?

Comment (Undin) [reply MAINTAINING.md]: I've added a couple of words about it into instruction.
Thanks

File: MAINTAINING.md [hunk 3]
```diff
@@ -49,6 +49,70 @@ Note `--distribution-type all`.
...
+#### How to support new platform version
+
+The following explanation uses `old`, `current` and `new` platform terms that mean:
+* `old` - number of the oldest supported platform version that should be dropped while
+* `current` - number of the latest major stable platform release
+* `new` - number of new platform that should be supported
+
+For example, at the moment of writing, `201` is `old`, `202` is `current` and `203` is `new`.
+ 
+Step to step instruction how to support new platform version:
```
Comment (ortem) [inline MAINTAINING.md:69]:
```suggestion
Step by step instruction on how to support new platform version:
```

File: MAINTAINING.md [hunk 4]
```diff
@@ -49,6 +49,70 @@ Note `--distribution-type all`.
...
+* Drop all code related to the `old` platform version, i.e. drop `gradle-%old%.properties` and
+all `%module_name%/src/%old%` directories in each module
+* Move code from `%module_name%/src/%current%` directories into common source set,
+i.e. `%module_name%/src/%current%/main` into `%module_name%/src/main` and `%module_name%/src/%current%/test` into `%module_name%/src/test`.
+Also, simplify the moved code if possible.
+* Add support of `new` platform, i.e. add `gradle-%new%.properties` with all necessary properties and make it compile.
```
Comment (ortem) [inline MAINTAINING.md:75]:
```suggestion
* Add support for `new` platform, i.e. add `gradle-%new%.properties` with all necessary properties and make it compile.
```

File: MAINTAINING.md [hunk 5]
```diff
@@ -49,6 +49,70 @@ Note `--distribution-type all`.
...
+It may be required to extract some code into platform-specific source sets to make plugin compile with each supported platform.
+See [Tips and tricks](#tips-and-tricks) section for the most common examples of how to do it
+* Fix tests
+* Update log path in run configurations like `runIDEA` and `runCLion`
```
Comment (ortem) [inline MAINTAINING.md:79]: Could you specify what exactly to update? Do I understand right that we should always have `idea-%current%.log` and `idea-%new%.log` here?

**Review by mchernyavsky — APPROVED** [review]

**Review by Kobzol — APPROVED** [review]

File: MAINTAINING.md (hunk 3 above)
Comment (Undin) [reply MAINTAINING.md:69]: Not sure about preposition here

File: MAINTAINING.md (hunk 5 above)
Comment (Undin) [reply MAINTAINING.md]: Added explanation

**Review by ortem — APPROVED** [review]

File: MAINTAINING.md [hunk 6]
```diff
@@ -49,6 +49,73 @@ Note `--distribution-type all`.
...
+* Update log path in run configurations like `runIDEA` and `runCLion`.
+For `runIDEA` configuration it requires:
+    - drop `idea-%old%.log` item 
+    - add `idea-%new%.log` item with `$PROJECT_DIR$/plugin/build/idea-sandbox-%new%/system/log/idea.log` path
+* Update CI workflows to use new platform version instead of old one.
+It usually requires just to update `platform-version` list in all workflows where we build the plugin.
+At the moment of writing they are `check.yml`, `rust-nightly.yml` and `rust-release.yml`
+* Fix `BACKCOMPAT: %old%` comments
+
+#### Tips and tricks
```
Comment (mchernyavsky) [inline MAINTAINING.md:88]: Maybe add a note about the `IgnoreInPlatform` annotation and `ApplicationInfo.getInstance().build`?

**Review by vlad20012 — COMMENTED** [review]
LGTM

**Review by vlad20012 — APPROVED** [review]

File: MAINTAINING.md (hunk 6 above)
Comment (Undin) [reply MAINTAINING.md:88]: Added

File: MAINTAINING.md [hunk 7]
```diff
@@ -49,6 +49,89 @@ Note `--distribution-type all`.
...
+Since we want to support at least the latest stable major platform release and new [EAP](https://www.jetbrains.com/resources/eap/)s,
+we have to compile the plugin with different platform versions to avoid any runtime errors.
+
+To solve the problem with incompatibilities, we use conditional compilation based on Gradle [source sets](https://docs.gradle.org/current/dsl/org.gradle.api.tasks.SourceSet.html).
+See [CONTRIBUTING.md](CONTRIBUTING.md#source-code-for-different-platform-versions) for more details about concrete source structure.
+
+#### How to support new platform version
+
+The following explanation uses `old`, `current` and `new` platform terms that mean:
+* `old` - number of the oldest supported platform version that should be dropped while
```
Comment (dima74) [inline MAINTAINING.md:63]:
```suggestion
* `old` - number of the oldest supported platform version that should be dropped
```
Or probably there is some missing word after `while`

File: MAINTAINING.md [hunk 8]
```diff
@@ -49,6 +49,89 @@ Note `--distribution-type all`.
...
+
+For example, at the moment of writing, `201` is `old`, `202` is `current` and `203` is `new`.
+ 
+Step by step instruction on how to support new platform version:
+* Drop all code related to the `old` platform version, i.e. drop `gradle-%old%.properties` and
+all `%module_name%/src/%old%` directories in each module
+* Move code from `%module_name%/src/%current%` directories into common source set,
+i.e. `%module_name%/src/%current%/main` into `%module_name%/src/main` and `%module_name%/src/%current%/test` into `%module_name%/src/test`.
+Also, simplify the moved code if possible.
+* Add support for `new` platform, i.e. add `gradle-%new%.properties` with all necessary properties and make it compile.
```
Comment (dima74) [inline MAINTAINING.md:75]: What is the best way to know build number to use? E.g. [currently it is `3645`](https://github.com/intellij-rust/intellij-rust/blob/f82f6cd68567e574bf1e30f5e0d263ee15d1d36e/gradle-203.properties#L1-L2)

**Review by dima74 — APPROVED** [review]

File: MAINTAINING.md (hunk 7 above)
Comment (Undin) [reply MAINTAINING.md]: Thanks

File: MAINTAINING.md (hunk 8 above)
Comment (Undin) [reply MAINTAINING.md:76]: I've added info into `gradle-%platform_version%.properties` files

**Pull request merged**